test-apps/appui-test-app/appui-test-providers/src/ui/providers/ComponentExamplesProvider.tsx

```diff
@@ -0,0 +1,756 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+/* eslint-disable deprecation/deprecation */
+
+import * as React from "react";
+import { Point } from "@itwin/core-react";
+import { ComponentExampleCategory, ComponentExampleProps } from "../frontstages/ComponentExamples";
+import { AccuDrawDialog, AccuDrawWidget, BackstageAppButton, BasicNavigationWidget, Calculator, CommandItemDef, ConfigurableCreateInfo,
+  ConfigurableUiContent, ContentControl, ContentDialog, ContentGroup, ContentLayout, ContentLayoutDef,
+  CursorInformation, CursorMenuData, CursorPopupContent, CursorPopupManager, CustomItemDef, ElementTooltip,
+  ExpandableSection, GroupItemDef, IModelConnectedViewSelector, KeyinEntry, KeyinPalettePanel, ListItem, ListItemType, ListPicker, ListPickerItem, MenuButton, MenuItemProps,
+  MessageCenterField, MessageManager, ModelessDialog, PositionPopup, PositionPopupContent, QuantityFormatSettingsPage,
+  SectionsStatusField, SplitPane, StandardMessageBox, StatusBar, StatusBarCenterSection,
+  StatusBarComposer, StatusBarDialog, StatusBarIndicator, StatusBarLabelIndicator,
+  StatusBarLeftSection, StatusBarRightSection, StatusBarSeparator, StatusBarSpaceBetween,
+  TileLoadingIndicator, ToolAssistanceField, ToolbarComposer, ToolbarHelper, ToolbarOrientation, ToolbarUsage,
+  ToolSettingsGridContainer, UiDataProvidedDialog, UiFramework, UiSettingsPage, UnitSystemSelector, ViewAttributesStatusField } from "@itwin/appui-react";
+import { MessageBoxIconType, MessageBoxType, NotifyMessageDetails, OutputMessagePriority,
+  OutputMessageType, QuantityType } from "@itwin/core-frontend";
+import { ConditionalBooleanValue, DialogButtonDef, DialogButtonType, DialogItem, DialogItemValue,
+  DialogLayoutDataProvider, DialogPropertyItem, DialogPropertySyncItem, PropertyChangeResult, PropertyChangeStatus, PropertyDescription, StandardContentLayouts, StandardTypeNames } from "@itwin/appui-abstract";
+import { StatusBarDialogTitleBar } from "@itwin/appui-react/lib/cjs/appui-react/statusbar/dialog/TitleBar";
+import { StatusBarDialogTitleBarButton } from "@itwin/appui-react/lib/cjs/appui-react/statusbar/dialog/Button";
+import { ComponentGenerator } from "@itwin/appui-react/lib/cjs/appui-react/uiprovider/ComponentGenerator";
+import { UnitSystemKey } from "@itwin/core-quantity";
+import { ToolbarItemContext } from "@itwin/components-react";
+import { Button } from "@itwin/itwinui-react";
+
+class TestContentControl extends ContentControl {
+  constructor(info: ConfigurableCreateInfo, options: any) {
+    super(info, options);
+
+    this.reactNode = <div>Test</div>;
+  }
+}
+
+class TestUiDataProvider extends DialogLayoutDataProvider {
```

Comment:
To do this, my understanding is that I would have to make a file with the classes in the appui package, export that, and the import in the test app. I am not sure that is the best idea considering we would only want this for use internally. I could import the file directly without exporting, but that would break the import within package lint rule which I could override. However, I am not sure that is the best idea. The way that I currently have this set up is parallel to how classes from test files were reused for component examples previously in the twins-core repo.

Reply:
I agree, let's avoid exporting test-specific things from our packages. It might indicate poor unit test quality if providers used for visual testing match our unit test providers.